rules/no-assign-parameter.js

```diff
@@ -20,47 +20,65 @@ module.exports = {
 
     create: function (context) {
 
-        function inspectBody (body, node, params) {
-            for (let statement of body) {
-                if (statement['type'] === 'ExpressionStatement') {
-
-                    // Check for any assignment of the variables
-                    if (statement['expression']['type'] === 'AssignmentExpression' &&
-                        params.indexOf(statement['expression']['left']['name']) >= 0) {
-                        context.report ({
-                            node: node,
-                            message: 'Disallow assigning to function parameters'
-                        });
-                    }
-
-                    // Check for incrementing and decrementing
-                    if (statement['expression']['type'] === 'UpdateExpression' &&
-                        params.indexOf(statement['expression']['argument']['name']) >= 0) {
-                        context.report ({
-                            node: node,
-                            message: 'Disallow assigning to function parameters'
-                        });
-                    }
-                }
+        function isBadAssignment (statement, params) {
+            return statement['expression']['type'] === 'AssignmentExpression' &&
+                   params.indexOf(statement['expression']['left']['name']) >= 0
+        }
 
-                if (['ForStatement', 'WhileStatement', 'DoWhileStatement'].indexOf(statement['type']) >= 0) {
-                    inspectBody (statement['body']['body'], node, params);
-                }
+        function isBadUpdate (statement, params) {
+            return statement['expression']['type'] === 'UpdateExpression' &&
+                   params.indexOf(statement['expression']['argument']['name']) >= 0
+
+        }
+
+        function checkExpressionStatement(statement, node, params) {
+            if(isBadAssignment(statement, params) || isBadUpdate(statement, params)) {
+                context.report({
+                    node: node,
+                    message: 'Disallow assigning to function parameters'
```

Comment:
Just to clarify, the message is different from description. The message is shown to the user. So here, the message should be `"<Function Name>": Avoid assigning to function parameters.`